src/booleans.js

```diff
@@ -1,62 +1,109 @@
 function negate(a) {
-  // your code here
-};
+  if (a === false) {
+    return true;
+  }
+  return false;
+}
 
 function both(a, b) {
-  // your code here
-};
+  if (a === true && b === true) {
+    return true;
+  }
+  return false;
+}
 
 function either(a, b) {
-  // your code here
-};
+  if (a === true || b === true) {
+    return true;
+  }
+  return false;
+}
 
 function none(a, b) {
-  // your code here
-};
+  if (a === false && b === false) {
+    return true;
+  }
+  return false;
+}
 
 function one(a, b) {
-  // your code here
-};
+  if ((a === true && b === false) || (a === false && b === true)) {
```

Comment:
Based on the above, I think you should be able to make this (and all the other functions in this file) more concise using the ideas above :)